lib/components/dashboard/course-list/Semester.tsx

```diff
@@ -260,6 +260,10 @@ const Semester: FC<{
       });
     } else {
       console.log('Failed to add', data.errors);
+      data.errors.forEach((error) => {
+        toast.error(error.detail);
+      });
```

Comment:
could this print stuff other than 

`Cannot take same course multiple times in the same semester`? 

I'm just scared that we'll have toasts with some crazy long error messages that are meaningless to users. As an example, we should not toast `500 internal error`. Please make sure the toast will only ever print the error message about duplicate course warning.

Reply:
hmm good point, I didn't think about that, the only way I can think of handling this is checking the error message, but I'm worried that might not be a great solution, any thoughts?

Reply:
how about checking status code? 
so in backend, reserve 400 for `Cannot take same course multiple times in the same semester` err and return 500 otherwise. See comment in your backend PR.

Reply:
Just made the changes!